scripts/dict_utils.py

```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env python
+
+import functools, operator
+
+# Copied from StackOverflow response
+def getFromDict(data, path):
```

Comment:
These util functions could have definitely used documentation on what an example input might look like.  But I like the functional programming syntax going on in your code, it makes things really succinct.